drivers: ethernet: Add timeout when waiting for LITEETH_TX_READY

mateusz-holenko opened the pull request:
This commit eliminates an inifite waiting for the TX ready flag
in the eth liteeth driver, exiting with error after a defined
number of attempts.

Signed-off-by: Pawel Sagan <psagan@antmicro.com>
Signed-off-by: Mateusz Holenko <mholenko@antmicro.com>

Commit:
drivers: ethernet: Add timeout when waiting for LITEETH_TX_READY

This commit eliminates an inifite waiting for the TX ready flag
in the eth liteeth driver, exiting with error after a defined
number of attempts.

Signed-off-by: Pawel Sagan <psagan@antmicro.com>
Signed-off-by: Mateusz Holenko <mholenko@antmicro.com>

jukkar commented on drivers/ethernet/eth_liteeth.c:
```diff
@@ -208,8 +218,11 @@ static void eth_iface_init(struct net_if *iface)
 	/* set MAC address */
-	net_if_set_link_addr(iface, context->mac_addr, sizeof(context->mac_addr),
-			     NET_LINK_ETHERNET);
+	if(net_if_set_link_addr(iface, context->mac_addr, sizeof(context->mac_addr),
+			     NET_LINK_ETHERNET) < 0) {
+		LOG_ERR("setting mac failed");
+		return;	
+	}
```
This looks like it is not part of "TX ready" change. Could you create a separate commit for this.

Commit:
drivers: ethernet: Verify returned value from net_if_set_link_addr

This commit verifies if the mac configuration is done correctly
during liteeth setup, and prompts a warning when not.

Signed-off-by: Pawel Sagan <psagan@antmicro.com>
Signed-off-by: Mateusz Holenko <mholenko@antmicro.com>

jukkar reviewed (approved):
LGTM

mateusz-holenko commented:
Can we merge this, or is there anything else I should do with this PR?

jukkar commented:
> Can we merge this, or is there anything else I should do with this PR?

Needs 1 more approver. @tbursztyka can you check this one?

pfalcon commented on drivers/ethernet/eth_liteeth.c:
```diff
@@ -96,7 +99,10 @@ static int eth_tx(const struct device *dev, struct net_pkt *pkt)
 	/* wait for the device to be ready to transmit */
 	while (sys_read8(LITEETH_TX_READY) == 0) {
-		;
+		if ((attempts++) == MAX_TX_FAILURE) {
```
I don't think we normally use parens around increment in such syntactic context in Zephyr.

mateusz-holenko replied:
I removed parnes as suggested

pfalcon commented:
> I removed parnes as suggested

Thanks.

@jukkar : This should be ready to merge.

jukkar commented:
> This should be ready to merge.

Yes, will do that after buildkite has finished.
